presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -330,14 +338,59 @@ private boolean checkTablePermission(ConnectorSecurityContext context, SchemaTab
         return false;
     }
 
+    private boolean checkTableAnyPermission(ConnectorSecurityContext context, SchemaTableName tableName)
+    {
+        if (INFORMATION_SCHEMA_NAME.equals(tableName.getSchemaName())) {
+            return true;
+        }
+        if (!canAccessSchema(context, tableName.getSchemaName(), AccessMode.READ_ONLY)) {
+            return false;
+        }
+        for (TableAccessControlRule rule : tableRules) {
+            Optional<Set<TablePrivilege>> tablePrivileges = rule.match(context.getIdentity().getUser(), tableName);
+            if (tablePrivileges.isPresent()) {
+                if (tablePrivileges.get().isEmpty()) {
+                    return false;
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+
     private boolean isSchemaOwner(ConnectorSecurityContext context, String schemaName)
     {
         for (SchemaAccessControlRule rule : schemaRules) {
-            Optional<Boolean> owner = rule.match(context.getIdentity().getUser(), schemaName);
+            Optional<Boolean> owner = rule.isSchemaOwner(context.getIdentity().getUser(), schemaName);
             if (owner.isPresent()) {
                 return owner.get();
             }
         }
         return false;
     }
+
+    private boolean canAccessSchema(ConnectorSecurityContext context, String schemaName, AccessMode requiredAccess)
+    {
+        for (SchemaAccessControlRule rule : schemaRules) {
+            Optional<AccessMode> accessMode = rule.match(context.getIdentity().getUser(), schemaName);
+            if (accessMode.isPresent()) {
+                return accessMode.get().implies(requiredAccess);
+            }
+        }
+        // if not rule match, we assume schema as READ_ONLY by default
+        return true;
```

Comment:
`return true;` doesn't match the comment above. Did you mean
```
return READ_ONLY.implies(requiredAccess);
```
?

Reply:
hmm...
 canAccessSchema == ` at least READ_ONLY `, right ?

Reply:
I am not sure I follow. This method takes `requiredAccessMode`.
The comment suggests "schema is at least READ_ONLY to user by default".
So the end line should be "return true but only if requiredAccessMode is READ_ONLY".

cc @kokosing